pkgs/applications/version-management/git-and-tools/git/default.nix

```diff
@@ -88,7 +90,7 @@ stdenv.mkDerivation {
 
   makeFlags = [
     "prefix=\${out}"
-    "SHELL_PATH=${stdenv.shell}"
+    "SHELL_PATH=${if (stdenv.hostPlatform != stdenv.buildPlatform) then "/bin/sh" else stdenv.shell}"
```

Comment:
```suggestion
    "SHELL_PATH=${runtimeShell}"
```

Reply:
Just read the commit message... The only pure alternative to your approach would be to replace `stdenv.shell` with `runtimeShell` in `$out`.

Reply:
Otherwise if we have a sandbox build `/bin/sh` is reasonable reproducible, could add the reasoning as a comment above for a future reader?

Reply:
We are referencing `"/bin/sh"` in nixpkgs … oh no.

Reply:
Yeah if it is about the old vs new version of the code (stateful) I like to put in the commit message, but if it is about the new version of the code alone, I rather it be a comment.

Comments have a prayer of being edited to stay up to date :), but commit messages only describe a moment in time.

Reply:
Yeah - patch-shebangs.sh bails out when the path is in the nix store.

I'll leave this change out of this set for now. Using /bin/sh is definitely not good, so some other fix is needed. Git will still build, you just can't run some of its shell scripts.

Reply:
:thinking: isn't this a general problem with cross builds, for e.g. Python install scripts?

How about using a substitution set for buildDeps that should be replaced with runDeps? And maybe automatically testing for buildDeps that are referenced in the output but not specifically allowed?